### PR TITLE
Turn on sam conversion by default [ci: last-only]

### PR DESCRIPTION
**Pull request by adriaanm**
Cleaned up while I was in the area (code & spec). Minimal spec for SAM conversion.
- [ ] setting to disable sam conversion
- [ ] check overloading spec, add test for ToString example to test/files/pos/sammy_overload.scala
- [ ] update comments in test/files/pos/t9449.scala
- [ ] more tests based on https://gist.github.com/lrytz/ec4da087dc65e421eb7d
- [ ] remove mention of <static> from spec

### CODE REVIEW

**Comment by adriaanm** [comment]
Review by @retronym

**Comment by adriaanm** [comment]
Review by @lrytz, bitte.

**Comment by lrytz** [comment]
Should we have a setting to turn the feature off?

File: test/files/pos/t9449.scala [hunk 1]
```diff
@@ -0,0 +1,19 @@
+trait II {
+  def apply(x: Int): Int
+}
+
+object Test {
+  def ii(x: Int): Int = x
+  def test = {
+    val ii1: II = x => ii(x) // works
+    val ii2: II = ii         // works (adapting `ii` to `II`)
+    val ii3: II = ii _       // fails -- should work
```
Comment (lrytz) [inline test/files/pos/t9449.scala:10]: it's a bit confusing to have the comment saying "fails", as it's actually fixed

**Comment by lrytz** [comment]
Should have some tests for overloading resolution. The following used to compile:

```
scala> abstract class A[T, R]{ def apply(x: T): R }
defined class A

scala> object O {
     |   def m(a: A[Int, Int]) = 0
     |   def m(a: Int => Int) = 1
     | }
defined object O

scala> O.m(x => x)
<console>:15: error: missing parameter type
       O.m(x => x)
           ^
```

I haven't looked at the details in overloading resolution that cause this change. Is this expected? I don't remember if we already talked about it. Maybe we can have a specific warning for when this happens?

**Comment by lrytz** [comment]
here's a few things i tried, can be included in a test: https://gist.github.com/lrytz/ec4da087dc65e421eb7d

**Comment by adriaanm** [comment]
incorporated into #4971

**Pull request closed without merging**